microcosm_pubsub/dispatcher.py

```diff
@@ -105,19 +127,24 @@ def invoke_handler(self, handler, media_type, content):
             )
             return handler_with_context(content)
```

Comment:
The `handler_with_context` is passed `self.message_context`... and invokes it with a dictionary.

I barely understand what this code is trying to do so I haven't fixed it yet.